src/touchstone/decision_maker/__init__.py

```diff
@@ -31,16 +31,23 @@ def _compare(self, input_dict, compare_dict):
         for u, v in input_dict.items():
             if u == self.baseline_uuid:
                 continue
-            metric_percent = v * 100 / input_dict[self.baseline_uuid]
-            # If percentage is greater than 100, sustract 100 from it else substract it from 100
-            deviation = metric_percent - 100 if metric_percent > 100 else 100 - metric_percent
-            deviation = -deviation if v < input_dict[self.baseline_uuid] else deviation
-            if (self.tolerancy >= 0 and v > base_val) or (self.tolerancy < 0 and v < base_val):
-                result = "Fail"
-                self.passed = False
-                self.fails += 1
-            else:
+            try:
+                metric_percent = v * 100 / input_dict[self.baseline_uuid]
+            # ZeroDivisionError means baseline value was 0, no comparison to be made here
+            except ZeroDivisionError:
                 result = "Pass"
+                deviation = 0
```

Comment:
should the deviation be set to 100% instead of 0 since it means baseline is having 0 value? Also we should have result based off tolerancy config similar to else block? may be use `finally:` block to reduce duplication

Reply:
Yes to the first, 100% would probably make more sense in most case, though if both values are 0 I'm not sure it would there (could always treat that specific case differently) 
As for your second point, I'm not sure how that'd look - here I'm just setting it to `Pass` so this doesn't unnecessarily fail any comparisons - did you have an idea for some kind of math we could use here for more dynamic result?

Reply:
> though if both values are 0 I'm not sure it would there (could always treat that specific case differently

right, that would be a corner case.

> did you have an idea for some kind of math we could use here for more dynamic result?
we could use same code as it exits in the `else` block for determining result based on tolerancy value, no? therefore I suggested to have a `finally:` block and move that code there.

Reply:
> > though if both values are 0 I'm not sure it would there (could always treat that specific case differently
> 
> right, that would be a corner case.
> 
I'm not seeing a way to check for both of those values 

> > did you have an idea for some kind of math we could use here for more dynamic result?
> 
> we could use same code as it exits in the `else` block for determining result based on tolerancy value, no? therefore I suggested to have a `finally:` block and move that code there.

But if we are setting the `deviation` value as part of the `Exception` block, what would the `finally` block be doing? As I understand it that block is calculating the deviation based on the `metric_percent` which is what is throwing this `ZeroDivisionError` exception